shared/p2p/adapter/metric/metric.go

```diff
@@ -0,0 +1,53 @@
+package metric
```

Comment:
A package level comment here would be useful for future readers and reference in Godoc as specified [here](https://blog.golang.org/godoc-documenting-go-code). Better function level godocs could also be useful with examples on how to use the adapter concretely.